### PR TITLE
fix warnings. avoid IterableOnce deprecated methods

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

File: src/library/scala/collection/immutable/Vector.scala [hunk 1]
```diff
@@ -202,12 +202,12 @@ final class Vector[+A] private[immutable] (private[collection] val startIndex: I
     // Implementation similar to `appendAll`: when of the collections to concatenate (either `this` or `prefix`)
     // has a small number of elements compared to the other, then we add them using `:+` or `+:` in a loop
     import Vector.{Log2ConcatFaster, TinyAppendFaster}
-    if (prefix.isEmpty) this
+    if (prefix.iterator.isEmpty) this
     else {
-      prefix.size match {
+      prefix.iterator.size match {
```
Comment (xuwei-k) [inline src/library/scala/collection/immutable/Vector.scala]: Maybe this changes fix https://github.com/scala/bug/issues/11122 by chance. But current implementation not efficient.

Comment (xuwei-k) [reply src/library/scala/collection/immutable/Vector.scala]: revert this changes. create another pull request https://github.com/scala/scala/pull/7153

File: src/library/scala/collection/immutable/LazyList.scala [hunk 2]
```diff
@@ -396,7 +396,7 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
 
   override def zip[B](that: collection.IterableOnce[B]): LazyList[(A, B)] =
     newLL {
-      if (this.isEmpty || that.isEmpty) State.Empty
+      if (this.isEmpty || that.iterator.isEmpty) State.Empty
```
Comment (NthPortal) [inline src/library/scala/collection/immutable/LazyList.scala]: you can leave these `LazyList` fixes out - they will be fixed in #7139

**Review by NthPortal — APPROVED** [review]
LGTM

**Pull request merged**